api-jersey/src/main/scala/com/yahoo/maha/api/jersey/MahaResource.scala

```diff
@@ -107,7 +107,9 @@ class MahaResource(mahaService: MahaService, baseRequest: BaseRequest) extends L
 
     val (reportingRequest: ReportingRequest, rawJson: Array[Byte]) = createReportingRequest(httpServletRequest, schemaOption.get, debug, forceEngine)
     val bucketParams: BucketParams = BucketParams(UserInfo(MDC.get(MahaConstants.USER_ID), Try(MDC.get(MahaConstants.IS_INTERNAL).toBoolean).getOrElse(false)), forceRevision = Option(forceRevision))
-    val mahaRequestProcessor: MahaRequestProcessor = MahaRequestProcessor(registryName, mahaService)
+
+    val defaultRequestCoordinator = DefaultRequestCoordinator(mahaService)
+    val mahaRequestProcessor: MahaRequestProcessor = MahaRequestProcessor(registryName, defaultRequestCoordinator, mahaService.mahaRequestLogWriter)
```

Comment:
Why not use the factory and create request with it?
```
  private[this] val mahaRequestProcessorFactory = MahaRequestProcessoryFactory(requestCoordinator, mahaService, mahaRequestLogWriter, mahaServiceMonitor)
```
Then just use it to create the processor:
```
  val mahaRequestProcessor: MahaRequestProcessor = mahaRequestProcessorFactory.create(registryName, processingLabel)
```